lib/query.js

```diff
@@ -2861,19 +2861,27 @@ Query.prototype.distinct = function(field, conditions) {
  * Cannot be used with `distinct()`
  *
  * @param {Object|String|Array<Array<(string | number)>>} arg
+ * @param {Object} [options]
+ * @param {Boolean} [options.override=false] If true, replace existing sort options with `arg`
  * @return {Query} this
  * @see cursor.sort https://www.mongodb.com/docs/manual/reference/method/cursor.sort/
  * @api public
  */
 
-Query.prototype.sort = function(arg) {
-  if (arguments.length > 1) {
-    throw new Error('sort() only takes 1 Argument');
+Query.prototype.sort = function(arg, options) {
+  if (arguments.length > 2) {
+    throw new Error('sort() takes at most 2 arguments');
+  }
+  if (options != null && typeof options !== 'object') {
+    throw new Error('sort() options argument must be an object or nullish');
   }
 
   if (this.options.sort == null) {
     this.options.sort = {};
   }
+  if (options && options.override) {
```

Comment:
In JS, somebody can pass `override: 1` or `override: '  '`, and even then this condition will pass. I think we should change the condition to `options && options.override === true`

Reply:
agreed

Also options.override is clearer at least than a boolean arg like in [`setOptions`](https://github.com/Automattic/mongoose/blob/master/lib/query.js#L1570)

Should we consider also adding this options.override for `.select()`? (I know there is `.projection()`, but it could simplify things to have just one method in the future, and maybe keep the other one as an alias, so same behavior)

Reply:
We prefer to treat `override: 1` or `override: '  '` as equivalent to `override: true`. There's 2 alternatives, either treat `override: 1` as `override: false` or throw an error; I don't think either is better in this case.

Re: adding `options.override` to `select()`, I think that's a reasonable idea to consider, but outside the scope of this issue. If this API works well, we'll consider adding it to `select()` and other query methods where it makes sense.